tools/speech_data_explorer/data_explorer.py

```diff
@@ -1,846 +1,1044 @@
-DATA_PAGE_SIZE = 10
-
-# operators for filtering items
-filter_operators = {
-    '>=': 'ge',
-    '<=': 'le',
-    '<': 'lt',
-    '>': 'gt',
-    '!=': 'ne',
-    '=': 'eq',
-    'contains ': 'contains',
-}
-
-# parse table filter queries
-def split_filter_part(filter_part):
-    for op in filter_operators:
-        if op in filter_part:
-            name_part, value_part = filter_part.split(op, 1)
-            name = name_part[name_part.find('{') + 1 : name_part.rfind('}')]
-            value_part = value_part.strip()
-            v0 = value_part[0]
-            if v0 == value_part[-1] and v0 in ("'", '"', '`'):
-                value = value_part[1:-1].replace('\\' + v0, v0)
-            else:
-                try:
-                    value = float(value_part)
-                except ValueError:
-                    value = value_part
-            return name, filter_operators[op], value
-    return [None] * 3
-
-
-# standard command-line arguments parser
-def parse_args():
-    parser = argparse.ArgumentParser(description='Speech Data Explorer')
-    parser.add_argument(
-        'manifest', help='path to JSON manifest file',
-    )
-    parser.add_argument('--vocab', help='optional vocabulary to highlight OOV words')
-    parser.add_argument('--port', default='8050', help='serving port for establishing connection')
-    parser.add_argument(
-        '--disable-caching-metrics', action='store_true', help='disable caching metrics for errors analysis'
-    )
-    parser.add_argument(
-        '--estimate-audio-metrics',
-        '-a',
-        action='store_true',
-        help='estimate frequency bandwidth and signal level of audio recordings',
-    )
-    parser.add_argument(
-        '--audio-base-path',
-        default=None,
-        type=str,
-        help='A base path for the relative paths in manifest. It defaults to manifest path.',
-    )
-    parser.add_argument('--debug', '-d', action='store_true', help='enable debug mode')
-    args = parser.parse_args()
-
-    # assume audio_filepath is relative to the directory where the manifest is stored
-    if args.audio_base_path is None:
-        args.audio_base_path = os.path.dirname(args.manifest)
-
-    print(args)
-    return args
-
-
-# estimate frequency bandwidth of signal
-def eval_bandwidth(signal, sr, threshold=-50):
-    time_stride = 0.01
-    hop_length = int(sr * time_stride)
-    n_fft = 512
-    spectrogram = np.mean(
-        np.abs(librosa.stft(y=signal, n_fft=n_fft, hop_length=hop_length, window='blackmanharris')) ** 2, axis=1
-    )
-    power_spectrum = librosa.power_to_db(S=spectrogram, ref=np.max, top_db=100)
-    freqband = 0
-    for idx in range(len(power_spectrum) - 1, -1, -1):
-        if power_spectrum[idx] > threshold:
-            freqband = idx / n_fft * sr
-            break
-    return freqband
-
-
-# load data from JSON manifest file
-def load_data(data_filename, disable_caching=False, estimate_audio=False, vocab=None, audio_base_path=None):
```

Comment:
Please integrate Ante's updates back (to support audio_base_path)

Reply:
done